frontend/src/model/Player.ts

```diff
@@ -0,0 +1,80 @@
+import Technologies from '../constants/technologies';
+import { defaultTechnology, defaultPlayersMeepleCount } from '../constants/gameDefaults';
+
+// temporary classes
+class Tile {}
+class Meeple {
+  player: Player;
+  constructor(player: Player) {
+    this.player = player;
+  }
+}
+// end of temporary classes
+
+class Player {
+  public name: string;
+  public technology: Technologies;
+  private score: number;
+  private placedTiles: Tile[];
+  private meepleList: Meeple[];
+
+  constructor(name: string, technology: Technologies = defaultTechnology) {
+    this.name = name;
+    this.technology = technology;
+
+    this.initializePlayerState();
+  }
+
+  initializePlayerState() {
+    this.score = 0;
+    this.placedTiles = [];
+    this.meepleList = [];
+
+    for (let i = 0; i < defaultPlayersMeepleCount; i++) {
+      this.meepleList.push(new Meeple(this));
+    }
+  }
+
+  computeScore(pointsToAdd: number = 0): number {
+    if (pointsToAdd >= 0) {
+      return (this.score += pointsToAdd);
+    } else {
+      console.log('Points to add should not be negative!');
+      return this.score;
+    }
+  }
+
+  updatePlacedTiles(tile: Tile): void {
```

Comment:
updatePlacedTile is too general - from code perspective it would be nice if we invoke player.placeTile

updatePlacedTile may be treated as update existing tile

Reply:
Done